web/src/components/apps/AppVersionHistory.jsx

```diff
@@ -1162,19 +1354,29 @@ class AppVersionHistory extends Component {
         handleViewLogs={this.handleViewLogs}
         handleSelectReleasesToDiff={this.handleSelectReleasesToDiff}
         renderVersionDownloadStatus={this.renderVersionDownloadStatus}
-        isDownloading={this.state.versionDownloadStatuses?.[version.sequence]?.downloadingVersion}
+        isDownloading={
+          this.state.versionDownloadStatuses?.[version.sequence]
+            ?.downloadingVersion
+        }
         adminConsoleMetadata={this.props.adminConsoleMetadata}
       />
     );
-  }
+  };
+
+  getPreflightState = (version) => {
+    let preflightState = "";
```

Comment:
method to getPreflightResultState, if it returns empty string. hasPreflightCheck will be set to false.
Hacky way of going about this, hasPreflightState will return 'fail' or 'pass'.. if empty we assume preflight checks spec does not exist. 
open to other ways of handling this.